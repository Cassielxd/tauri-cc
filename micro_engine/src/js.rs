// Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.

use deno_core::Snapshot;
use log::debug;

#[cfg(not(feature = "__runtime_js_sources"))]
static CLI_SNAPSHOT: &[u8] = include_bytes!(concat!(env!("OUT_DIR"), "/CLI_SNAPSHOT.bin"));

pub fn deno_isolate_init() -> Option<Snapshot> {
  debug!("Deno isolate init with snapshots.");
  #[cfg(not(feature = "__runtime_js_sources"))]
  {
    Some(Snapshot::Static(CLI_SNAPSHOT))
  }
  #[cfg(feature = "__runtime_js_sources")]
  {
    None
  }
}
