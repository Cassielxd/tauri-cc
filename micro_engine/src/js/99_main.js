// Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.
import "ext:cli/40_testing.js";
import "ext:cli/runtime/js/99_main.js";
