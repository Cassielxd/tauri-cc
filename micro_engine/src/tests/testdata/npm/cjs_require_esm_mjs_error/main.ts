import "npm:@denotest/cjs-require-esm-error/require_mjs.js";
