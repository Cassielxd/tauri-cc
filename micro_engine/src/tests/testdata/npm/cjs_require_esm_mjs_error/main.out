error: Uncaught Error: require() of ES Module [WILDCARD]esm_mjs.mjs from [WILDCARD]require_mjs.js not supported. Instead change the require to a dynamic import() which is available in all CommonJS modules.
    [WILDCARD]
